addon/components/gavant-table.ts

```diff
@@ -0,0 +1,532 @@
+import Component from '@ember/component';
+import { or, empty, not, and, gt, readOnly } from '@ember/object/computed';
+import { set, computed, get, setProperties, action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { scheduleOnce } from '@ember/runloop';
+import { A } from '@ember/array';
+import { tryInvoke, isEmpty } from '@ember/utils';
+import { htmlSafe } from '@ember/string';
+import { SafeString } from '@ember/template/-private/handlebars';
+import { lt, gte, conditional } from 'ember-awesome-macros';
+import ResizeAware from 'ember-resize/mixins/resize-aware';
+import Media from 'ember-responsive';
+import { observes } from '@ember-decorators/object';
+import NativeArray from '@ember/array/-private/native-array';
+// @ts-ignore: Ignore import of compiled template
+import layout from '../templates/components/gavant-table';
+import { TableColumn } from 'table-components';
+
+class GavantTableComponent extends Component.extend({ ResizeAware }) {
```

Comment:
If we can I'd probably prefer we keep this component named just `TableComponent`, we generally haven't prefixed any of our components in other addons with `Gavant`. And the single word `<Table>` component instantiation is just so clean :-D